OSDOCS#6030: cert-manager: setting log levels (#59777)
openshift/openshift-docs: modules/cert-manager-enable-operand-log-level.adoc

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operand-log-level_{context}"]
= Setting a log level for cert-manager

You can set a log level for cert-manager to determine the verbosity of log messages.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.

> * The {cert-manager-operator} 1.11.1 or later is installed.

Reviewer: Just a suggestion that you can consider: Is it possible to make this active voice?
E.g.
```suggestion
* You have installed the {cert-manager-operator} 1.11.1 or later.
```
If not, that's fine.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager.operator cluster
----

. Set the log level value by editing the `spec.logLevel` section:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
...
spec:
  logLevel: Normal <1>
----

> <1> The default `logLevel` is `Normal`. Replace `Normal` with the desired log level value. The valid log level values for the `CertManager` resource are `Normal`, `Debug`, `Trace`, and `TraceAll`. You can set `logLevel` to `Normal` to audit logs and perform common operations when everything is fine. To troubleshoot a minor issue, you can set `logLevel` to `Debug` to view verbose logs. To troubleshoot a major issue, you can set `logLevel` to `Trace` to view more verbose logs. To troubleshoot serious issues, you can set `logLevel` to `TraceAll`.

Reviewer: ```suggestion
<1> The default `logLevel` is `Normal`. Replace `Normal` with the desired log level value. The valid log level values for the `CertManager` resource are `Normal`, `Debug`, `Trace`, and `TraceAll`. You can set `logLevel` to `Normal` to audit logs and perform common operations when everything is fine. You can set `logLevel` to `Debug` to view verbose logs for troubleshooting a minor issue. You can set `logLevel` to `Trace` to view more verbose logs for troubleshooting a major issue. You can set `logLevel` to `TraceAll` for troubleshooting serious issues.
```

OR

```suggestion
<1> The default `logLevel` is `Normal`. Replace `Normal` with the desired log level value. The valid log level values for the `CertManager` resource are `Normal`, `Debug`, `Trace`, and `TraceAll`. To audit logs and perform common operations when everything is fine, set `logLevel` to `Normal` . To troubleshoot a minor issue by viewing verbose logs, set `logLevel` to `Debug` . To troubleshoot a major issue by viewing more verbose logs, you can set `logLevel` to `Trace`. To troubleshoot serious issues, set `logLevel` to `TraceAll`.
```

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operand-log-level_{context}"]
= Setting a log level for cert-manager

> You can set a log level for cert-manager to determine the amount of detail in logged error messages.

Reviewer 2: This one also needs update like Swarup commented here https://github.com/openshift/openshift-docs/pull/59777#discussion_r1190691164 .
```suggestion
You can set a log level for cert-manager to determine the verbosity of log messages.
```

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* The {cert-manager-operator} is installed.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager.operator cluster
----

. Set the log level value by editing the `spec.logLevel` section:
+
[source,yaml]
----

> apiVersion: operator.openshift.io/v1alpha1
> kind: CertManager
> metadata:
>   creationTimestamp: "2023-05-10T07:34:15Z"
>   finalizers:
>   - cert-manager-operator.operator.openshift.io/cert-manager-controller-deployment
>   - cert-manager-operator.operator.openshift.io/cert-manager-webhook-deployment
>   - cert-manager-operator.operator.openshift.io/cert-manager-cainjector-deployment
>   generation: 1
>   name: cluster
>   resourceVersion: "47290"
>   uid: 0f5cd145-a20f-43d1-ac3d-0db28a6f9f06

Reviewer 2: Fields creationTimestamp, resourceVersion, uid, et al ... are not needed to be shown.
```suggestion
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
...
```

spec:
  logLevel: Normal <1>
----

> <1> Replace `<log_level>` with the desired log level value. The default `logLevel` is `Normal`. The valid log level values for the `CertManager` resource are `Normal`, `Debug`, `Trace`, and `TraceAll`. You can set `logLevel` to `Normal` to audit logs and perform common operations when everything is fine. To troubleshoot a minor issue, you can set `logLevel` to `Debug` to view verbose logs. To troubleshoot a major issue, you can set `logLevel` to `Trace` to view more verbose logs. To troubleshoot serious issues, you can set `logLevel` to `TraceAll`.

Reviewer 2: Didn't see `log_level` placeholder.
```suggestion
<1> Replace `Normal` with the desired log level value. The default `logLevel` is `Normal`. The valid log level values for the `CertManager` resource are `Normal`, `Debug`, `Trace`, and `TraceAll`. You can set `logLevel` to `Normal` to audit logs and perform common operations when everything is fine. To troubleshoot a minor issue, you can set `logLevel` to `Debug` to view verbose logs. To troubleshoot a major issue, you can set `logLevel` to `Trace` to view more verbose logs. To troubleshoot serious issues, you can set `logLevel` to `TraceAll`.
```

+
[NOTE]
====
`TraceAll` generates huge amount of logs. After setting `logLevel` to `TraceAll`, you might experience performance issues.
====

> . Save the file to apply the changes.

Reviewer 2: It does oc edit to save changes for a resource, not usual sense of save a `file` (though oc edit uses temporary file)
```suggestion
. Save your changes and quit the text editor to apply your changes.
```

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-log-levels.adoc

:_content-type: PROCEDURE
[id="cert-manager-enable-operand-log-level_{context}"]
= Setting a log level for the cert-manager controller

You can set a log level for the cert-manager controller to determine the amount of detail in logged error messages.

> [NOTE]

Reviewer 3: Probably, if we want to add such a section, it'd be good to put descriptions of each log level type via details mentioned in https://github.com/openshift/api/blob/894b49f57a15cbce3869961e20cd9d52df6f8b0f/operator/v1/types.go#L97-L110
